src/Swift.Bindings/src/Model/MethodDecl.cs

```diff
@@ -0,0 +1,32 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+namespace BindingsGeneration
+{
+    /// <summary>
+    /// Represents a method declaration.
+    /// </summary>
+    public sealed record MethodDecl
```

Comment:
Using `record`s in public APIs hasn't been confirmed as yet. This is likely okay for the current state. Just be aware that if this is going to be publicly consumable API review may push back.